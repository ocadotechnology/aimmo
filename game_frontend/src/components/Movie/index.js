import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class Movie extends PureComponent {
  render () {
    const movie = this.props.movie
    return (
      <li>
        {movie.title} - {movie.director} - {movie.producer} - { movie.release_date}
      </li>
    )
  }
}

Movie.propTypes = {
  movie: PropTypes.shape({
    title: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    producer: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
  })
}
