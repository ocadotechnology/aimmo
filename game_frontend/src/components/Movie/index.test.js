/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Movie from 'components/Movie'
import renderer from 'react-test-renderer'

const singleMovie = {
  title: 'Test Title',
  director: 'Test Director',
  producer: 'Test Producer',
  release_date: '20/03/2018'
}

describe('<Movie />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Movie movie={singleMovie} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
