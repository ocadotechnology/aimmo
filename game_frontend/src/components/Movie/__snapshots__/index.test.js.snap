// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movie /> renders correctly 1`] = `
<li>
  Test Title
   - 
  Test Director
   - 
  Test Producer
   - 
  20/03/2018
</li>
`;
